src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -891,6 +891,9 @@ protected static function check_for_circular_dependencies( $dependents, $depende
 	 * @return string The plugin's slug.
 	 */
 	private static function convert_to_slug( $plugin_file ) {
+		if ( 'hello.php' === $plugin_file ) {
+			return 'hello-dolly';
```

Comment:
What about user added single file plugins?

Reply:
Current, the single file plugin `my-plugin.php` will be slugified to `my-plugin`.

Reply:
To further clarify, when Hello Dolly is upgraded and put into a directory, it becomes `hello-dolly/hello.php`, rather than `hello/hello.php`, so it would break without this special handling.

Dependents of Hello Dolly (should anyone ever get adventurous) would need to use `Requires Plugins: hello-dolly`, rather than `Requires Plugins: hello`, which makes sense since the latter would break as soon as someone installs/upgrades Hello Dolly from the Plugins Repository.